modules/nf-core/taxonkit/name2taxid/main.nf

```diff
@@ -0,0 +1,51 @@
+process TAXONKIT_NAME2TAXID {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/taxonkit:0.15.1--h9ee0642_0':
+        'biocontainers/taxonkit:0.15.1--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), val(name), path(names_txt)
```

Comment:
I see why you would want the input like this, but I'm not sure if it follows the guidelines since the actual input is a file.

Reply:
The file could be easily made using the `.collectFile()` operator

Reply:
I was just following the docs https://bioinf.shenwei.me/taxonkit/usage/#name2taxid. 

Although how do you mean with `collectFile()`. The output of that is `file` and not `[ meta, file ]`.

Reply:
Yes true, but that's nothing some channel logic can't handle :).

@maintainers what are your opinions on this?

Reply:
It's not particularly easy for a new developer though:
```nextflow
    Channel.of( [ id: 'test', taxid: 1234 ] )
        .tap { ch_data }
        .collectFile( newLine: true ) { meta -> [ "${meta.id}.tsv", "$meta.taxid" ] }
        .map{ file -> [ file.baseName , file ] }
        .join( ch_data.map { meta -> [ meta.id, meta ] }, by: 0 )
        .map{ id, idfile, meta -> [ meta, idfile ] }
```

Reply:
> @maintainers what are your opinions on this?

I think you pinged the wrong team. Hovering over the name on GH points to a non-nf-core team.

Reply:
@nf-core/maintainers it should be thanks for pointing out :grin:

Reply:
Yeah I don't mind this, but it clashes with the guideline of only allowing mandatory value inputs

Reply:
The value is mandatory if there's no file ( and I don't plan on generating that file )

Reply:
What is the difference between `name` and `names_txt`? As in you can supply a string versus a file?

Reply:
Could have two separate optional input channels, one for string only and one for file (but are mutually exclusive)? 

Might look alittle ugly, would be relatively easy for for a pipeline dev to put an `[]` in the channel they don't want with a (multi)Map?

Reply:
I think that's just adding unnecessary complexity to have to use `multiMap` to make this input.
Either way the input will likely need to be formed using a `map`